File: play-redis/src_play25-/main/java/play/cache/SyncCacheApi.java
package play.cache;

import java.util.concurrent.Callable;

public interface SyncCacheApi extends CacheApi {

    <T> T getOrElseUpdate(String key, Callable<T> block, int expiration);

    <T> T getOrElseUpdate(String key, Callable<T> block);

    @Override
    default <T> T getOrElse(String key, Callable<T> block, int expiration) {
        return this.getOrElseUpdate(key, block, expiration);
    }

    @Override
    default <T> T getOrElse(String key, Callable<T> block) {
        return this.getOrElseUpdate(key, block);
    }

}

File: play-redis/src_play25-/main/scala/play/api/cache/SyncCacheApi.scala
package play.api.cache

import scala.concurrent.duration.Duration
import scala.reflect.ClassTag

trait SyncCacheApi extends CacheApi {

  def getOrElseUpdate[A: ClassTag](key: String, expiration: Duration)(orElse: => A): A

  override final def getOrElse[A: ClassTag](key: String, expiration: Duration)(orElse: => A): A =
    getOrElseUpdate(key, expiration)(orElse)

}
